src/nist.py

```diff
@@ -66,4 +67,84 @@ def get_NIST_ids():
+        '7': {'16': '243.77', '18': '241.84','20': '240.86', '22': '240.75', '24': '241.31', '26': '242.45'},
+        '6': {'16': '343.00', '18': '341.53','20': '341.58', '22': '342.58', '24': '344.23', '26': '346.67'},
+        '5': {'16': '483.91', '18': '482.91','20': '484.97', '22': '486.92', '24': '490.24', '26': '494.55'},
+        '4': {'16': '675.07', '18': '686.88','20': '690.08', '22': '695.01', '24': '701.06', '26': '709.48'},
+        '3': {'16': '950.71', '18': '963.56','20': '987.27', '22': '1000.81', '24': '1015.70', '26': '1030.78'},
+        '2': {'16': '1274.07', '18': '1317.71','20': '1330.16', '22': '1355.29', '24': '1367.79', '26': '1395.94'},
+        '1': {'16': '1766.68', '18': '1830.34','20': '1883.97', '22': '1937.34', '24': '1987.50', '26': '2066.95'}
+        };
+    
+    ##Get keys and values of dictionary to construct a 2D array##
+    #Get dictionary keys as lists
+    sphereID = list(phantom_v2.keys());
+    temperature = list(phantom_v2['1'].keys());
+    
+    #Define array (data) to store data for interpolation
+    data = np.empty([len(temperature),len(sphereID)+1]);
+    #Fill data array
+    row = 0;
+    for i in temperature:
+        row = row + 1;
+        data[row-1,0] = int(i);
+        for j in sphereID:
+            T1_value = float(phantom_v2[j][i]);
+            data[row-1,int(j)] = T1_value;
+
+    ##Code for Temperature Correction: Interpolation##
+    #If input_temperature is a scalar:
+    #Validate input_temperature
+    validTemperature = lambda input_temperature: np.isscalar(input_temperature);
+    if validTemperature(input_temperature) == True:
+        input2array = input_temperature;
+        input_temperature = np.arange(1);
+        input_temperature[0] = input2array;
+    
+    
+    #Cubic Spline (define output as array and dictionary)
+    spline_estimatedT1 = np.empty([len(input_temperature),len(sphereID)]);
+    spline_estimatedT1_dictionary = {};
+    
+    #Cubic (define output as array and dictionary)
+    cubic_estimatedT1 = np.empty([len(input_temperature),len(sphereID)]);
+    cubic_estimatedT1_dictionary = {};
+    
+    #Interpolations
+    for k in range(len(sphereID)):
+        for l in range(len(input_temperature)):
+            #Cubic Spline
+            cubicSpline = interpolate.splrep(data[:,0], data[:,k+1]);
+            spline_estimatedT1[l,k] = interpolate.splev(input_temperature[l],cubicSpline);
+            spline_estimatedT1_dictionary[l,k+1] = spline_estimatedT1[l,k];
+            
+            #Cubic        
+            cubic = interpolate.interp1d(data[:,0], data[:,k+1], kind='cubic');
+            cubic_estimatedT1[l,k] = cubic(input_temperature[l]);
+            cubic_estimatedT1_dictionary[l,k+1] = cubic_estimatedT1[l,k];
+    
+
+        
+    return spline_estimatedT1_dictionary, spline_estimatedT1, cubic_estimatedT1_dictionary, cubic_estimatedT1;
```

Comment:
Instead of outputting each of these seperately, we should only be outputting one array of T1s. Now, which one of these could instead be selected using a function argument (with the one we prefer set as a default). 

Also, the format of the output of this function (i.e. type and size) should match exactly the output of the function get_reference_NIST_values above.

Reply:
@mathieuboudreau the user can now introduce the kind of interpolation (I still have to understand why `cubic` and `cubic-spline` are giving the same results) with `cubic-spline` as default. The output now is one array of T1 values. 

The format of the output is exactly the same as the output of the function `get_reference_NIST_values` if the input temperature is 20°C. That is, an array of size (14,1) with the T1s in the rows.